app/controllers/avo/search_controller.rb

```diff
@@ -72,6 +61,31 @@ def search_resource(resource)
       [resource.name.pluralize.downcase, result_object]
     end
 
+    # Figure out if it's a belongs to search and if it has the attach_scope block present.
+    # If so, set the parent for those edit screens and the grandparent for those new screens.
+    def apply_attach_scope(query)
+      return query if params[:via_reflection_class].blank?
+
+      # Fetch the field
+      field = belongs_to_field
+
+      # No need to modify the query if there's no attach_scope present.
+      return query if field.attach_scope.blank?
+
+      # Try to fetch the parent.
+      if params[:via_reflection_id].present?
+        parent = params[:via_reflection_class].safe_constantize.find params[:via_reflection_id]
+      end
+
+      # Try to fetch the grandparent for the new views where the parent is nil.
+      if params[:via_parent_resource_id].present? && params[:via_parent_resource_class].present?
+        grandparent = params[:via_parent_resource_class].safe_constantize.find params[:via_parent_resource_id]
+      end
```

Comment:
Hmmm...I wonder if it would be feasible to use `find_or_initialize_by` to pass in the association on the parent vs. introducing another variable? Would look something like:

```ruby
grandparent = params[:via_parent_resource_class].safe_constantize.find params[:via_parent_resource_id]

parent = params[:via_reflection_class].safe_constantize.find_or_initialize_by(
  "id" => params[:via_reflection_id],
  params[:via_parent_resource_relation] => grandparent
)
```

I'm not familiar enough with Avo's internals to know where the grandparent relation might come from, so not really sure on the variable naming...

Another thought might be to use [the `GlobalID` feature](https://github.com/rails/globalid/blob/main/README.md) introduced to Rails alongside ActiveJob to serialize the parent into a URI perhaps?

Reply:
Hey @davidlormor.

TL;DR: Try `2.6.1.pre.2` and use `parent.industry` in the attach scope. You'll have access to the `Industry` record.

Long version👇

In the scenario where you'd edit the `AdminAccount`, and try to change the `belongs_to` of the `person`, the parent is a record of `AdminAccount` class. So, in the same manner, when creating an `AdminAccount`, when setting the `person`, the `parent` will still have to be an `AdminAccount` instance, but there isn't one right now because you're just creating it.

I followed your suggestion and initialized the parent model `AdminAccount` with the grandparent attached. So in the attach scope you can use `parent.industry`. `parent` will be an `AdminAccount` without an ID, but with the `industry_id` attribute populated so you have all the info needed to scope the query.

I fixed the `searchable: false` bug too. Thanks for the shoutout.